File: src/backend/src/config/env.js
require("dotenv").config();

module.exports = {
  port: process.env.PORT || 3000,

  mqtt: {
    host: process.env.MQTT_HOST || "localhost",
    port: parseInt(process.env.MQTT_PORT || "1883", 10),
    username: process.env.MQTT_USERNAME || "",
    password: process.env.MQTT_PASSWORD || "",
  },

  influx: {
    url: process.env.INFLUX_URL || "http://localhost:8086",
    org: process.env.INFLUX_ORG || "EcoGuard",
    bucket: process.env.INFLUX_BUCKET || "sensors",
    token: process.env.INFLUX_TOKEN || "",
  },

  env: process.env.NODE_ENV || "development",
};

File: src/backend/src/config/env.py
import os
from typing import Optional
from pydantic_settings import BaseSettings


class MQTTSettings(BaseSettings):
    host: str = "localhost"
    port: int = 1883
    username: str = ""
    password: str = ""

    class Config:
        env_prefix = "MQTT_"


class InfluxSettings(BaseSettings):
    url: str = "http://localhost:8086"
    org: str = "EcoGuard"
    bucket: str = "sensors"
    token: str = ""

    class Config:
        env_prefix = "INFLUX_"


class Settings(BaseSettings):
    port: int = 3000
    cors_origin: str = "*"
    debug: bool = False
    env: str = "development"

    mqtt: MQTTSettings = MQTTSettings()
    influx: InfluxSettings = InfluxSettings()

    class Config:
        env_file = ".env"
        env_nested_delimiter = "__"


# Singleton instance
settings = Settings(
    mqtt=MQTTSettings(),
    influx=InfluxSettings()
)